app/models/team.rb

```diff
@@ -567,23 +567,32 @@ def search_for_similar_articles(query, pm = nil)
     # query:  expected to be text
     # pm: to request a most relevant to specific item and also include both FactCheck & Explainer
     limit = pm.nil? ? CheckConfig.get('most_relevant_team_limit', 3, :integer) : CheckConfig.get('most_relevant_item_limit', 10, :integer)
-    result_ids = Bot::Smooch.search_for_similar_published_fact_checks_no_cache('text', query, [self.id], limit).map(&:id)
-    items = []
-    unless result_ids.blank?
-      # I depend on FactCheck to filter result instead of report_design
-      items = FactCheck.where(report_status: 'published')
-      .joins(claim_description: :project_media)
-      .where('project_medias.id': result_ids)
-      # Exclude the ones already applied to a target item if exsits
-      items = items.where.not('fact_checks.id' => pm.fact_check_id) unless pm&.fact_check_id.nil?
-    end
-    if items.blank? || !pm.nil?
-      # Get Explainers if no fact-check returned or get similar_articles for a ProjectMedia
-      ex_items = Bot::Smooch.search_for_explainers(nil, query, self.id, limit)
+    threads = []
+    fc_items = []
+    ex_items = []
+    threads << Thread.new {
+      result_ids = Bot::Smooch.search_for_similar_published_fact_checks_no_cache('text', query, [self.id], limit, nil, nil, nil, false).map(&:id)
+      unless result_ids.blank?
+        fc_items = FactCheck.joins(claim_description: :project_media).where('project_medias.id': result_ids)
+        if pm.nil?
+          # This means we obtain relevant items for the Bot preview, so we should limit FactChecks to published articles;
+          # otherwise, relevant articles for ProjectMedia should include all FactChecks.
+          fc_items = fc_items.where(report_status: 'published')
+        elsif !pm.fact_check_id.nil?
+          # Exclude the ones already applied to a target item if exists.
+          fc_items = fc_items.where.not('fact_checks.id' => pm.fact_check_id) unless pm&.fact_check_id.nil?
```

Comment:
@melsawy does this need a distinct as well, or were the duplicates only coming from explainers?

Reply:
Only for Explainers.